lib/convection/control/stack.rb

```diff
@@ -586,12 +582,39 @@ def get_events(pages = nil, stack_name = id)
         end
       rescue Aws::CloudFormation::Errors::ValidationError # Stack does not exist
       rescue Aws::EC2::Errors::RequestExpired
-        raise unless @credential_error_count <= @credential_error_max_retries
+        raise if reached_max_credential_retries?
+
+        wait_for_credential_refresh
+
+        reload_cf_client
+        retry
+      end
+
+      def client_options
+        client_options = { region: @region }
+        client_options[:credentials] = @credentials unless @credentials.nil?
 
+        client_options
+      end
+
+      def reload_cf_client
+        Aws.shared_config.fresh if @credentials.nil?
```

Comment:
1. This needs to continue to work with IAM user credentials, have you verified that?
2. Why do we need to refresh the credentials?  Can we create a new Client obj?  Can we let the user refresh the credentials themselves?

It seems like credentials expiring in the middle of the command shouldn't affect the outcome.  That is the command should be immutable.